serialization/python_pickle.ksy

```diff
@@ -0,0 +1,585 @@
+            'opcode::long': decimalnl_long
+            'opcode::long1': long1
+            'opcode::long4': long4
+            'opcode::string': stringnl
+            'opcode::binstring': string4
+            'opcode::short_binstring': string1
+            'opcode::binbytes': bytes4
+            'opcode::short_binbytes': bytes1
+            'opcode::binbytes8': bytes8
+            'opcode::none': no_arg
+            'opcode::newtrue': no_arg
+            'opcode::newfalse': no_arg
+            'opcode::unicode': unicodestringnl
+            'opcode::short_binunicode': unicodestring1
+            'opcode::binunicode': unicodestring4
+            'opcode::binunicode8': unicodestring8
+            'opcode::float': floatnl
+            'opcode::binfloat': f8
+            'opcode::empty_list': no_arg
+            'opcode::append': no_arg
+            'opcode::appends': no_arg
+            'opcode::list': no_arg
+            'opcode::empty_tuple': no_arg
+            'opcode::tuple': no_arg
+            'opcode::tuple1': no_arg
+            'opcode::tuple2': no_arg
+            'opcode::tuple3': no_arg
+            'opcode::empty_dict': no_arg
+            'opcode::dict': no_arg
+            'opcode::setitem': no_arg
+            'opcode::setitems': no_arg
+            'opcode::empty_set': no_arg
+            'opcode::additems': no_arg
+            'opcode::frozenset': no_arg
+            'opcode::pop': no_arg
+            'opcode::dup': no_arg
+            'opcode::mark': no_arg
+            'opcode::pop_mark': no_arg
+            'opcode::get': decimalnl_short
+            'opcode::binget': u1
+            'opcode::long_binget': u4
+            'opcode::put': decimalnl_short
+            'opcode::binput': u1
+            'opcode::long_binput': u4
+            'opcode::memoize': no_arg
+            'opcode::ext1': u1
+            'opcode::ext2': u2
+            'opcode::ext4': u4
+            'opcode::global':  stringnl_noescape_pair
+            'opcode::stack_global': no_arg
+            'opcode::reduce': no_arg
+            'opcode::build': no_arg
+            'opcode::inst': stringnl_noescape_pair
+            'opcode::obj': no_arg
+            'opcode::newobj': no_arg
+            'opcode::newobj_ex': no_arg
+            'opcode::proto': u1
+            'opcode::stop': no_arg
+            'opcode::frame': u8
+            'opcode::persid': stringnl_noescape
+            'opcode::binpersid': no_arg
+        doc: |
+          Optional argument for the operation. Data type and length
+          are determined by the value of the opcode.
+
+  decimalnl_short:
+    seq:
+      - id: val
+        type: str
+        encoding: ascii
+        terminator: 0x0a # "\n"
+    doc: |
+      Integer or boolean, encoded with the ASCII characters [0-9-].
+
+      The values '00' and '01' encode the Python values `False` and `True`.
+      Normally a value would not contain leading '0' characters.
+
+  decimalnl_long:
+    seq:
+      - id: val
+        type: str
```

Comment:
Shouldn't they be `strz`? Does the compiler accept them? @GreyCat, isn't it a bug?

Reply:
The values are definitely not terminated by `\0`. In general Python doesn't use NUL terminated strings, and `\0` is allowed anywhere in a `bytes`, or `str`/`unicode` object.

Reply:
`strz` is for any terminated strings, `\0` is just the default terminator. `str` is for strings with explicitly specified size.

Reply:
Okay, but please be aware that's not the impression I got from http://doc.kaitai.io/ksy_reference.html#_strings

> To designate attribute as string type, use `type: str` and provide encoding info [...] `type: strz` can be also used as a shortcut to define a null-terminated string (C-style). I.e. these foo and bar attributes are equivalent:
> 
> ```
> - id: foo
>   type: str
>   terminator: 0
> - id: bar
>   type: strz
> ```

Reply:
you are completely right and I was wrong: `strz` is just an alias to `str` with zeto terminator:

https://github.com/kaitai-io/kaitai_struct_compiler/blob/21ec2e7b91b5e2f68465e74813f20b1da4fd5d8c/shared/src/main/scala/io/kaitai/struct/datatype/DataType.scala#L347L351

So don't change the stuff, it is OK in the current state.